hyp3_gamma/insar/unwrapping_geocoding.py

```diff
@@ -97,24 +108,63 @@ def get_neighbors(array, i, j, n=1):
     return array[i_start:i_stop, j_start:j_stop]
 
 
-def ref_point_with_max_cc(data_cc: np.array, window_size=1):
-    '''
-    shift determine the window size, n=1 9-pixel window, n=2, 25-pixel window, etc.
-    '''
+def find_ref_point_with_largest_cc(data_cc: np.ndarray, indices: (np.array, np.array),
+                                   window_size, start_idx, pick_num):
 
-    data_cc_max = data_cc[data_cc < 1.0].max()
-    idx = np.where(data_cc == data_cc_max)
+    # get the indices of the  pixels with the highest coherence values
+    rows = indices[0][start_idx:start_idx + pick_num]
+
+    cols = indices[1][start_idx:start_idx + pick_num]
 
-    rows, cols = idx[0], idx[1]
     num = len(rows)
     tots = np.zeros(num, dtype=float)
 
     for k in range(num):
-        tots[k] = get_neighbors(data_cc, rows[k], cols[k], window_size).sum()
+        neighbors = get_neighbors(data_cc, rows[k], cols[k], window_size)
+        if (neighbors == 0.0).any() or neighbors.size < (2*window_size+1)**2:
+            tots[k] = 0.0
+        else:
+            tots[k] = neighbors.sum()
+
+    if (tots == 0.0).all():
+        ref_i = None
+        ref_j = None
+    else:
+        idx = np.where(tots == tots.max())
+        ref_i = rows[idx[0][0]]
+        ref_j = cols[idx[0][0]]
+
+    return ref_i, ref_j
+
+
+def ref_point_with_max_cc(data_cc: np.array, window_size=10, pick_num=20, cc_thresh=0.3):
+    """window size determines the actual window, for example, n=1 means 9-pixel window, n=2 means 25-pixel window, etc.
+    Args:
+        data_cc: array includes cc data
+        window_size: defines the actual window, the actual window is a square of 2*window_size + 1
```

Comment:
I think this would be more clear.
```suggestion
    """
    Args:
        data_cc: array includes cc data
        window_size: the actual window is a square of size (2*window_size + 1); for example, window_size=1 means 9-pixel window, window_size=2 means 25-pixel window, etc.
```